pkg/iniparser_test.go

```diff
@@ -0,0 +1,1308 @@
+	value         string
+	expectedError error
+}
+
+func sliceEqual(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	a_copy := make([]string, len(a))
+	b_copy := make([]string, len(b))
+
+	copy(a_copy, a)
+	copy(b_copy, b)
+
+	sort.Strings(a_copy)
+	sort.Strings(b_copy)
+
+	return reflect.DeepEqual(a_copy, b_copy)
+}
+
+func TestINI_LoadFromString(t *testing.T) {
+	parser := IniFile{}
+	emptyMap := make(map[string]map[string]string)
+	testCases := []LoadFromStringTestCase{
+		{
+			desc:          "Empty string as input",
+			input:         "",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc:          "Only one comment line start with # as input",
+			input:         "#This is a comment",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc:          "Only one comment line start with ; as input",
+			input:         ";This is a comment",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc:          "Only one key value line as input",
+			input:         "key = value",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc:          "Only one section line as input",
+			input:         "[section]",
+			expectedError: nil,
+			expectedMap:   map[string]map[string]string{"section": make(map[string]string)},
+		},
+		{
+			desc: "Only comments as input",
+			input: "!This is a comment 1\n " +
+				"!This is a comment 2\n" +
+				"!This is a comment 3\n" +
+				"!This is a comment 4",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc: "Only sections as input",
+			input: "[section1]\n" +
+				"[section2]\n" +
+				"[section3]\n" +
+				"[section4]",
+			expectedError: nil,
+			expectedMap: map[string]map[string]string{"section1": make(map[string]string),
+				"section2": make(map[string]string),
+				"section3": make(map[string]string),
+				"section4": make(map[string]string)},
+		},
+		{
+			desc: "Only keys values as input",
+			input: "key1 = value1\n " +
+				"key2 = value2\n" +
+				"key3 = value3\n" +
+				"key4 = value4",
+			expectedError: errFileIsEmpty,
+			expectedMap:   emptyMap,
+		},
+		{
+			desc: "Normal case 1",
+			input: `;This is a comment1
+;This is a comment2
+
+[section1]
+key1 = value1
+key2 = value2`,
+			expectedError: nil,
+			expectedMap: map[string]map[string]string{"section1": {
+				"key1": "value1",
+				"key2": "value2",
+			},
+			},
+		},
+		{
+			desc: "Normal case 2",
+			input: `;This is a comment1
+;This is a comment2
+
+[section1]
+key1 = value1
+key2 = value2
+key3 = value3
+key4 = value4
+
+[section2]
+key1 = value1
+key2 = value2`,
+			expectedError: nil,
+			expectedMap: map[string]map[string]string{"section1": {
+				"key1": "value1",
+				"key2": "value2",
+				"key3": "value3",
+				"key4": "value4",
+			},
+				"section2": {
+					"key1": "value1",
+					"key2": "value2",
+				},
+			},
+		},
+		{
+			desc: "Normal case 3",
```

Comment:
does adding more section affect the behavior of the parser? if no why testing the behavior then?